File: contractCalls/getClusterInfo.py
#!/usr/bin/env python3

import sys

def getClusterInfo(clusterAddress, eBlocBroker=None, web3=None): 
    if eBlocBroker is None and web3 is None: 
        import os
        from imports import connectEblocBroker, getWeb3
        web3        = getWeb3()
        eBlocBroker = connectEblocBroker(web3)

    clusterAddress = web3.toChecksumAddress(clusterAddress) 
    
    if str(eBlocBroker.functions.isClusterExist(clusterAddress).call()) == "False":
        print("Cluster(" + clusterAddress + ") is not registered. Please try again with registered Ethereum Address as cluster.")
        sys.exit() 

    clusterPriceInfo = eBlocBroker.functions.getClusterInfo(clusterAddress).call()
    blockReadFrom      = clusterPriceInfo[0]
    availableCoreNum   = clusterPriceInfo[1]    
    priceCoreMin       = clusterPriceInfo[2]
    priceDataTransfer  = clusterPriceInfo[3]
    priceStorage       = clusterPriceInfo[4]
    priceCache         = clusterPriceInfo[5]
    commitmentBlockNum = clusterPriceInfo[6]

    my_filter = eBlocBroker.eventFilter('LogClusterInfo', {'fromBlock': int(blockReadFrom) - 100, 'toBlock': int(blockReadFrom) + 1})
    '''
    my_filter = eBlocBroker.events.LogCluster.createFilter(
        fromBlock=int(blockReadFrom),       
        toBlock=int(blockReadFrom) + 1,
        argument_filters={'clusterAddress': str(clusterAddress)}
    )    
    loggedJobs = my_filter.get_all_entries()
    print(loggedJobs[0])
    sp'''
    # print(my_filter.get_all_entries()[0])
    
    return('{0: <20}'.format('blockReadFrom: ')     + str(blockReadFrom)  + '\n' +
           '{0: <20}'.format('availableCoreNum: ')  + str(availableCoreNum) + '\n' +
           '{0: <20}'.format('priceCoreMin: ')      + str(priceCoreMin)  + '\n' +
           '{0: <20}'.format('priceDataTransfer: ') + str(priceDataTransfer)  + '\n' +
           '{0: <20}'.format('priceStorage: ')      + str(priceStorage)  + '\n' +
           '{0: <20}'.format('priceCache: ')        + str(priceCache)  + '\n' +
           '{0: <20}'.format('commitmentBlockNum: ')  + str(commitmentBlockNum) + '\n' +
           '{0: <20}'.format('clusterEmail: ')      + my_filter.get_all_entries()[0].args['clusterEmail'] + '\n' +
           '{0: <20}'.format('miniLockID: ')        + my_filter.get_all_entries()[0].args['miniLockID'] + '\n' +
           '{0: <20}'.format('ipfsAddress: ')       + my_filter.get_all_entries()[0].args['ipfsAddress'] + '\n' +
           '{0: <20}'.format('fID: ')               + my_filter.get_all_entries()[0].args['fID'] + '\n' +
           '{0: <20}'.format('whisperPublicKey: ')  + '0x' + my_filter.get_all_entries()[0].args['whisperPublicKey']);

if __name__ == '__main__':
    if len(sys.argv) == 2:
        clusterAddress = str(sys.argv[1]) 
    else:        
        clusterAddress = "0x4e4a0750350796164D8DefC442a712B7557BF282"

    print(getClusterInfo(clusterAddress))

File: contractCalls/getClusterInfo.js
var eBlocBroker = require('../contract.js');

Web3 = require("web3");
web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));

if(!web3.isConnected()){
    console.log("notconnected");
    process.exit();
}

if (process.argv.length == 3)
    clusterAddress = process.argv[2]
else
    clusterAddress = "0x75a4c787c5c18c587b284a904165ff06a269b48c"

var myContractInstance  = web3.eth.contract(eBlocBroker.abi).at(eBlocBroker.address);
fromBlock = myContractInstance.getDeployedBlockNumber();

var array = myContractInstance.getClusterInfo(clusterAddress);
var eBlocBrokerEvent = myContractInstance.LogCluster({}, {fromBlock: array[0], toBlock: 'latest'});

eBlocBrokerEvent.watch( function (error, result) {
    if(result == null){
        fs.appendFile( myPath, "notconnected", function(err) {
            process.exit();
        });
        eBlocBrokerEvent.stopWatching()
    }

    var cluster = result.args.cluster;
    var clusterName;
    var fID;
    var miniLockId; 
    var ipfsAddress;
    
    if (cluster == clusterAddress) {
	clusterName = result.args.clusterName;
	fID         = result.args.fID;
	miniLockId  = result.args.miniLockId;
	ipfsAddress = result.args.ipfsAddress;	
    }
    console.log("clusterName: "        + clusterName);
    console.log("fID: "                + fID);
    console.log("miniLockId: "         + miniLockId);
    console.log("ipfsAddress: "        + ipfsAddress);
    console.log("coreNumber: "         + array[1]);
    console.log("corePriceMinuteWei: " + array[2]);   
    process.exit();	    
});

